linkerd/app/outbound/src/zone.rs

```diff
@@ -0,0 +1,52 @@
+use crate::metrics::prom;
+use linkerd_app_core::{
+    metrics::OutboundZoneLocality, svc, transport::metrics::zone::TcpZoneMetrics,
+};
+use prometheus_client::encoding::EncodeLabelSet;
+
+#[derive(Clone, Debug, Default)]
+pub struct TcpZoneMetricsParams {
+    bytes: prom::Family<TcpZoneLabels, prom::Counter>,
+}
+
+impl TcpZoneMetricsParams {
+    pub fn register(registry: &mut prom::Registry) -> Self {
+        let bytes = prom::Family::default();
+        registry.register(
+            "bytes",
+            "The total amount of read and written bytes to this proxy",
+            bytes.clone(),
+        );
```

Comment:
We probably want to use [register_with_unit](https://docs.rs/prometheus-client/latest/prometheus_client/registry/struct.Registry.html#method.register_with_unit) to enforce best practices (including unit comment metadata). This will also force to choose a more narrowly descriptive/disambiguating name...

Off the top of my head, maybe something like `transfer_cost`, which will give us something like `outbound_tcp_transfer_cost_bytes_total`.

To that end, do we need to separate the receive and send sides of this to accurately reflect cost? I imagine that upload and download are not necessarily priced identically. We could reflect that with op=Recv and op=Send labels, or we could use distinct metrics -- the former being more amenable to aggregation when we do not have distinct upload/download costs.

Reply:
Good call on the unit and naming, the name was definitely something that needed refinement.

As far as separating out the metrics, I think one metric with labels is better for aggregation as you mention. I'd imagine symmetric costs is the more common scenario, while still leaving the option for separating them out.